cloudant-sync-datastore-core/src/test/java/com/cloudant/sync/internal/mazha/CouchConfig.java

```diff
@@ -34,33 +39,98 @@
     // a reverse proxy
     private URI rootUri;
 
+    private String username;
+
+    private String password;
+
     private List<HttpConnectionRequestInterceptor> requestInterceptors = new ArrayList
             <HttpConnectionRequestInterceptor>();
     private List<HttpConnectionResponseInterceptor> responseInterceptors = new ArrayList
             <HttpConnectionResponseInterceptor>();
 
     public CouchConfig(URI rootUri) {
-        this.rootUri = rootUri;
+        this(rootUri,
+                Collections.<HttpConnectionRequestInterceptor>emptyList(),
+                Collections.<HttpConnectionResponseInterceptor>emptyList(),
+                null,
+                null);
     }
 
     public CouchConfig(URI rootUri,
                        List<HttpConnectionRequestInterceptor> requestInterceptors,
-                       List<HttpConnectionResponseInterceptor> responseInterceptors) {
+                       List<HttpConnectionResponseInterceptor> responseInterceptors,
+                       String username,
+                       String password) {
         this.rootUri = rootUri;
         this.requestInterceptors = requestInterceptors;
         this.responseInterceptors = responseInterceptors;
+        this.username = username;
+        this.password = password;
+
     }
 
-    public List<HttpConnectionRequestInterceptor> getRequestInterceptors() {
+
+    public List<HttpConnectionRequestInterceptor> getRequestInterceptors(boolean includeCookie) {
+        CookieInterceptor cookieInterceptor = buildCookieInterceptor();
```

Comment:
I'm not greatly in favour of a getter building an interceptor. I know this is test code, but it is pretty confusing to have an API with a setter and a getter, where the getter can do something other than return what was set in the setter.